src/librustdoc/clean/mod.rs

```diff
@@ -2577,7 +2584,9 @@ impl<'tcx> Clean<Type> for Ty<'tcx> {
                 let mut n = cx.tcx.lift(&n).unwrap();
                 if let ConstVal::Unevaluated(def_id, substs) = n.val {
                     let param_env = cx.tcx.param_env(def_id);
-                    n = cx.tcx.const_eval(param_env.and((def_id, substs))).unwrap()
+                    if let Ok(new_n) = cx.tcx.const_eval(param_env.and((def_id, substs))) {
+                        n = new_n;
+                    }
```

Comment:
Can you use the same trick in the other function? You just have to do this first, before `const_eval`:
```rust
let n = tcx.mk_const(ty::Const {
    val: ConstVal::Unevaluated(def_id, substs),
    ty: tcx.types.usize
});
```

Reply:
Do you mean *instead* of `const_eval`? Otherwise, this new assignement to `n` will never be used...

Reply:
I mean it'd be this in the other place (line 2450) - rewrote it a bit since previous comment:
```rust
let n = cx.tcx.const_eval(param_env.and((def_id, substs))).unwrap_or_else(|_| {
    cx.tcx.mk_const(ty::Const {
        val: ConstVal::Unevaluated(def_id, substs),
        ty: cx.tcx.types.usize
    })
});
```